### ISSUE TITLE
MariaDB 10.3 compatibility in regard to mb4

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#29483]
@VicDeo mentioned that MariaDB 10.3 will get rid of the innodb table format switch.

The question is whether this will make the DB default to always use mb4 or not.

We'll need to:
1) find a alternative for the missing config switch
2) if MariaDB 10.3 defaults to mb4 or to the other table format, check whether we need to provide a migration for all the data.

Let's do the research first before deciding on what to do.

@DeepDiver1975 @VicDeo

**Comment by VicDeo** [comment]
Meanwhile MariaDB 10.3.3 with **Beta** status was released on 23 Dec 2017.
IMO this is something for 10.1

**Comment by PVince81** [comment]
Agreed, moving to planned.

Do you have a rough idea about how much work this will be ? (estimate)

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `status/STALE` is set and there were no updates for 7 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).)

**Comment by VicDeo** [comment]
We're getting closer....
https://mariadb.com/kb/en/library/release-notes-mariadb-103-series/
MariaDB 10.3.5 
**Status: RC** | Release Date: 26 Feb 2018

**Comment by PVince81** [comment]
moving to planned

with a bit of luck no changes are needed on the code and only testing needed?

**Comment by VicDeo** [comment]
I wouldn't count on bit of gluck here... We had unit tests with 10.3 when it was in alpha and they failed.

To revive the essence of  the issue:
`innodb_file_format` option is used to detect utfmb4 here 
https://github.com/owncloud/core/blob/92b658a60e6c28ef9dd7594384180e0d73e96660/lib/private/DB/MySqlTools.php#L35-L36

According to https://mariadb.com/kb/en/library/xtradbinnodb-server-system-variables/#innodb_file_format 
it's always barracuda since MariaDB 10.0.2.2
```
Deprecated: MariaDB 10.2
Removed: MariaDB 10.3.1
```

**Comment by VicDeo** [comment]
so, for MariaDB 10.3 it's enough to bypass checking this option.
But there could be more surprises of a different kind.

**Comment by PVince81** [comment]
something to look into for OC 11 while we upgrade libs and everything

**Comment by PVince81** [comment]
@VicDeo any action points ? if yes, would be good to estimate and we can put this in the next sprint.

Might need to wait for https://github.com/owncloud/core/pull/32065

**Comment by VicDeo** [comment]
@PVince81 Adding 10.3.8 to the build matrix would be a good start to confirm that issue persists.
Estimate is up to 0.5md.  (optimistic: 1-2 hrs)

Basically it needs  omiting `'innodb_file_format' => 'Barracuda'` check  in case `Platform instanceof MariaDb1027Platform` (10.3 has no it's own platform and this options is always barracuda since MariaDB 10.2.2) 
Estimates for the tests are less clear :)

**Comment by PVince81** [comment]
added 10.3 to the build matrix here: https://github.com/owncloud/core/pull/33255

**Comment by PVince81** [comment]
> But there could be more surprises of a different kind.

@VicDeo So far none have been detected in our test suites.

Anything that comes to mind ? If we want to play it safe we could leave this update to be OC 11 only.

**Comment by phil-davis** [comment]
Reopening because this support never got released. The changes "died" with the old `master` that is `archive-master`

They are now waiting in #36289 for us to drop PHP 7.0 support

**Comment by phil-davis** [comment]
`mariadb:10.3` support has now been merged to `master` planned for release in core `10.4`
`phpunit` is running and passing with `mariadb:10.3`
https://drone.owncloud.com/owncloud/core/21589/15/10
```
OK, but incomplete, skipped, or risky tests!
Tests: 9901, Assertions: 55811, Skipped: 77.
```

Now I will try a run of acceptance tests with `mariadb:10.3` database backend.

**Comment by phil-davis** [comment]
PR #36493 runs all unit and acceptance tests with mariadb:10.3 and passes.
So this is good - closing the issue, "fixed" by #36290
